src/ray/common/bundle_spec.cc

```diff
@@ -144,6 +144,40 @@ std::string GetOriginalResourceNameFromWildcardResource(const std::string &resou
   return match_groups[1].str();
 }
 
+std::optional<PgFormattedResourceData> ParsePgFormattedResource(
+    const std::string &resource, bool for_wildcard_resource, bool for_indexed_resource) {
```

Comment:
Could we use the above API calls instead of adding a new one?

Reply:
The existing API cannot handle non-wildcard resources. There's `GetOriginalResourceName`, but this API assumes we already know the resources are pg resources. I can change this in this PR and fix, but I feel like the risk is pretty low...